cpp/src/arrow/flight/meson.build

```
@@ -44,7 +44,25 @@ install_headers(
     subdir: 'arrow/flight',
 )
 
-grpc_dep = dependency('grpc++')
+grpc_opts = []
+if cpp_compiler.get_id() == 'msvc'
+    grpc_opts += [
+        'openssl:asm=disabled',  # asm cannot compile on Windows
+        're2:default_library=static',  # re2 does not export symbols
+    ]
+endif
+
+# Due to the Meson bug in https://github.com/mesonbuild/meson/issues/14905
+# we need to special how the grpc_cpp_plugin resolution is handled
+# when found on the system versus when built as a subproject
+grpc_cpp_dep = dependency('grpc++', allow_fallback: false, required: false)
+if grpc_cpp_dep.found()
+    has_system_grpc = true
+else
+    grpc_cpp_dep = dependency('grpc++')
+    has_system_grpc = false
+endif
```

Can we simplify this?

```suggestion
has_system_grpc = grpc_cpp_dep.found()
if not has_system_grpc
    grpc_cpp_dep = dependency('grpc++')
endif
```